internal/wallet/wallet_service.go

```diff
@@ -0,0 +1,184 @@
+package wallet
+
+import (
+	"encoding/json"
+	"net/http"
+
+	model "github.com/o-mercan/Wallet-Service-Api/internal/model"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type Service struct {
+	DB *gorm.DB
+}
+
+type WalletService interface {
+	AddNewWallet(wallet model.Wallet) (model.Wallet, error)
+	DepositWallet(ID uint, newWallet model.Wallet) (model.Wallet, error)
+	WithdrawWallet(ID uint, newWallet model.Wallet) (model.Wallet, error)
+	GetCurrentBalance(ID uint) (model.Wallet, error)
+	GetTransactionsReport()
+	////////////////////
+	GetWalletByID(ID uint) (model.Wallet, error)
+	PostUser(user model.User) (model.User, error)
+	GetUsers() (*[]model.User, error)
+	PostWallet(wallet model.Wallet) (model.Wallet, error)
+	GetWallets() (*[]model.Wallet, error)
+}
+
+func (s *Service) AddNewWallet(wallet model.Wallet) (model.Wallet, error) {
+	//result := s.DB.Create(wallet)
+	//s.DB.Session(&gorm.Session{FullSaveAssociations: true}).Updates(&wallet)
+	//return wallet, result.Error
+	s.DB.Session(&gorm.Session{FullSaveAssociations: true}).Updates(&wallet)
+	if result := s.DB.Create(&wallet); result.Error != nil {
+		log.WithFields(log.Fields{
+			"Function": "AddNewWallet",
+		}).Error(result.Error.Error())
+		return model.Wallet{}, nil
+	}
+
+	return wallet, nil
+
+}
+
+func (s *Service) DepositWallet(ID uint, newWallet model.Wallet) (model.Wallet, error) {
+	var wallet model.Wallet
+	wlt, err := s.GetWalletByID(ID)
+	if err != nil {
+		return model.Wallet{}, err
+	}
+	depositAmount := newWallet.Amount
+	if result1 := s.DB.Model(&wlt).Update("amount", gorm.Expr("amount + ?", depositAmount)); result1.Error != nil {
+		log.WithFields(log.Fields{
```

Comment:
loglanmasına ihtiyac yok. Loglar hataları ve varsa önemli gördüğümüz yerler için kullanırız